store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java

```diff
@@ -231,10 +248,11 @@ protected void reset(Mailbox mbox, ZimbraSoapContext zsc) throws ServiceExceptio
     }
 
     private static void validateEmail(String email, Account account) throws ServiceException {
-        String[] addresses = account.getMailAddress();
-        if (Arrays.asList(addresses).contains(email)) {
+        String accountName = account.getName();
+        String aliases[] = account.getAliases();
+        if (Arrays.asList(aliases).contains(email) || accountName.equals(email)) {
             throw ServiceException
```

Comment:
Do you want to log at the debug level what the 2 email addresses were.

Reply:
Done